cf-agent/verify_files_utils.c

```diff
@@ -1345,20 +1345,22 @@ bool CopyRegularFile(EvalContext *ctx, const char *source, const char *dest, con
         free(tmpstr);
     }
     else
-    {
 #endif
-
+    {
         strlcpy(new, dest, CF_BUFSIZE);
 
         if (!JoinSuffix(new, sizeof(new), CF_NEW))
         {
             Log(LOG_LEVEL_ERR, "Unable to construct filename for copy");
             return false;
         }
+    }
 
-#ifdef __APPLE__
+    bool dest_exists = false;
+    if (stat(dest, &dest_stat) == 0)
+    {
+        dest_exists = true;
```

Comment:
I'd do `ret = stat(dest, &dest_stat); dest_exists = (ret == 0)` here because `dest_stat` is used later and is not just populated as a side effect of checking if the file exists or not.